### PR TITLE
Feature/fetch projects

### PR DESCRIPTION
**Pull request by bbdsphe**

### CODE REVIEW

**Review by bryce-bbd — APPROVED** [review]

**Comment by bryce-bbd** [comment]
Looks good 

But bro I see why it took so long XD. Can you explain what the modifier does?

**Review by batandwa-bbd — CHANGES_REQUESTED** [review]
I don't understand why you had to do this. It had nothing to do with Listing Projects?

**Comment by umlayezi** [comment]
> I don't understand why you had to do this. It had nothing to do with Listing Projects?

It helps with IDE functionality. It was suggested that we use a single solution file.

**Comment by umlayezi** [comment]
> Looks good
> 
> But bro I see why it took so long XD. Can you explain what the modifier does?

It just stores the query parameters sent with the url (for e.g. <URL>/project?search=...&sort=desc). The database eventually uses these parameters to paginate and filter the results.

File: server/src/Controllers/ProjectsController.cs [hunk 1]
```diff
@@ -25,9 +26,17 @@ public class ProjectsController(ProjectsService projects) : ControllerBase
         private readonly ProjectsService projects = projects;
 
         [HttpGet]
-        public IActionResult Fetch()
-        {
-            return Ok(projects);
+        public async Task<IActionResult> Fetch(string? search, string? orderBy, string? sort, int? size, int? page) {
```
Comment (batandwa-bbd) [inline server/src/Controllers/ProjectsController.cs]: Can you update this to only fetch projects for the current user. Not all projects. Just current user projects.

**Review by batandwa-bbd — CHANGES_REQUESTED** [review]

File: server/src/Controllers/ProjectsController.cs (hunk 1 above)
Comment (umlayezi) [reply server/src/Controllers/ProjectsController.cs]: I'll work on that next.

Comment (batandwa-bbd) [reply server/src/Controllers/ProjectsController.cs]: Alright. Pls make it part of this pull request.

**Pull request closed without merging**